packages/node/test/integrations/undici.test.ts

```diff
@@ -0,0 +1,319 @@
+    fetch = require('undici').fetch;
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.warn('Undici integration tests are skipped because undici is not installed.');
+  }
+});
+
+const DEFAULT_OPTIONS = getDefaultNodeClientOptions({
+  dsn: SENTRY_DSN,
+  tracesSampleRate: 1,
+  integrations: [new Undici()],
+});
+
+beforeEach(() => {
+  const client = new NodeClient(DEFAULT_OPTIONS);
+  hub = new Hub(client);
+  makeMain(hub);
+});
+
+afterEach(() => {
+  requestHeaders = {};
+  setTestServerOptions({ statusCode: 200 });
+});
+
+afterAll(() => {
+  getTestServer()?.close();
+});
+
+conditionalTest({ min: 16 })('Undici integration', () => {
+  it.each([
+    [
+      'simple url',
+      'http://localhost:18099',
+      undefined,
+      {
+        description: 'GET http://localhost:18099/',
+        op: 'http.client',
+      },
+    ],
+    [
+      'url with query',
+      'http://localhost:18099?foo=bar',
+      undefined,
+      {
+        description: 'GET http://localhost:18099/',
+        op: 'http.client',
+        data: {
+          'http.query': '?foo=bar',
+        },
+      },
+    ],
+    [
+      'url with POST method',
+      'http://localhost:18099',
+      { method: 'POST' },
+      {
+        description: 'POST http://localhost:18099/',
+      },
+    ],
+    [
+      'url with POST method',
+      'http://localhost:18099',
+      { method: 'POST' },
+      {
+        description: 'POST http://localhost:18099/',
+      },
+    ],
+    [
+      'url with GET as default',
+      'http://localhost:18099',
+      { method: undefined },
+      {
+        description: 'GET http://localhost:18099/',
+      },
+    ],
+  ])('creates a span with a %s', async (_: string, request, requestInit, expected) => {
+    const transaction = hub.startTransaction({ name: 'test-transaction' }) as Transaction;
+    hub.getScope().setSpan(transaction);
+
+    await fetch(request, requestInit);
+
+    expect(transaction.spanRecorder?.spans.length).toBe(2);
+
+    const span = transaction.spanRecorder?.spans[1];
+    expect(span).toEqual(expect.objectContaining(expected));
+  });
+
+  it('creates a span with internal errors', async () => {
+    const transaction = hub.startTransaction({ name: 'test-transaction' }) as Transaction;
+    hub.getScope().setSpan(transaction);
+
+    try {
+      await fetch('http://a-url-that-no-exists.com');
+    } catch (e) {
+      // ignore
+    }
+
+    expect(transaction.spanRecorder?.spans.length).toBe(2);
+
+    const span = transaction.spanRecorder?.spans[1];
+    expect(span).toEqual(expect.objectContaining({ status: 'internal_error' }));
+  });
+
+  it('does not create a span for sentry requests', async () => {
+    const transaction = hub.startTransaction({ name: 'test-transaction' }) as Transaction;
+    hub.getScope().setSpan(transaction);
+
+    try {
+      await fetch(`${SENTRY_DSN}/sub/route`, {
+        method: 'POST',
+      });
```

Comment:
l (and more a question than a suggestion): 
This makes me wonder about our `isSentryRequest` check. As I mentioned yesterday, we have 5 slightly different checks for identifying a sentry request all over the code base. I'll sooner or later clean this up (after #7626) but should this check take the Http method into account? Some checks do ([example](https://github.com/getsentry/sentry-javascript/blob/e1af1e6b9b781d55a10ac9df7b7090a59056d895/packages/browser/src/integrations/breadcrumbs.ts)) while others (like [the one](https://github.com/getsentry/sentry-javascript/blob/e1af1e6b9b781d55a10ac9df7b7090a59056d895/packages/node/src/integrations/utils/http.ts) you're using here) don't. I'd argue it's probably not necessary to check for the method but wdyt?

Reply:
All sentry requests use POST atm, so I don't see anything wrong with this for now, but the more future-proof and correct solution is to not take into account the http method (in case we do ever introduce `GET` methods - think something like dynamic config).

Going to keep this as is for the test, we can come back and refactor all instances of `isSentryRequest` later.